code/src/app/unitenew/uniteDirectives/template.selector.ts

```diff
@@ -18,8 +18,10 @@ export class TemplateSelector {
 
     renderTemplate(value)
     {
-        let siteConfig = this._glConfig.getGlobalConfig('site');
-        let componentFactory = this._cfResolver.resolveComponentFactory(value[siteConfig.template]);
-        this._vcRef.createComponent(componentFactory);
+        //@Todo - Need to improve this code
+        this._glConfig.getGlobalConfig('site').subscribe(data => {
+            let componentFactory = this._cfResolver.resolveComponentFactory(value[data.template]);
```

Comment:
@ttplsudhir  - Add condition before this line to check if the template is available inside `value`

Reply:
@sushan05 Yes will fix it.

Reply:
@sushan05 Done this , Please check.